### PR TITLE
ci: introduce Rust workflow

### PR DESCRIPTION
**Pull request by qrayven**
add GitHub workflows

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      - run: cargo check
+
+  security:
+    name: Dependencies security audit
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cancel previous runs
+        uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ github.token }}
```
Comment (shumkov) [inline .github/workflows/rust.yml]: Can be handled with built-in workflow functionality https://docs.github.com/en/actions/using-jobs/using-concurrency

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -0,0 +1,112 @@
+on:
+  workflow_dispatch:
+  pull_request:
+
+name: rs-dpp
+
+jobs:
+  wasm-build:
```
Comment (shumkov) [inline .github/workflows/rust.yml]: I would rename it to `wasm-errors` for naming consistency.

File: .github/workflows/rust.yml [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          default: true
+          components: rustfmt
+
+      - run: exit `cargo fmt --check | wc -l`
+
+  errors:
```
Comment (shumkov) [inline .github/workflows/rust.yml]: I would rename it `rust-errors` for naming consistency

File: .github/workflows/rust.yml [hunk 4]
```diff
@@ -0,0 +1,112 @@
+on:
+  workflow_dispatch:
+  pull_request:
+
+name: rs-dpp
```
Comment (shumkov) [inline .github/workflows/rust.yml]: This is a workflow name option, which represents specific workflow not the entire repo. So I would keep the same name as you called the file: `Rust`.

**Review by shumkov — CHANGES_REQUESTED** [review]

**Comment by shumkov** [comment]
Should we move other changes unrelated to CI to a separate PR?

**Comment by qrayven** [comment]
> Should we move other changes unrelated to CI to a separate PR?

It was intended. But the pipeline wouldn't pass if those changes are not included. Or maybe you mean making the PR fixing pipeline first and then another PR with the pipeline in itself?

**Comment by shumkov** [comment]
Yeah... I guess right now is not important to keep PR with specific atomic changes, but eventually, it would be nice to follow this practice.

**Review by shumkov — APPROVED** [review]
LGTM! 👍

**Pull request merged**